crates/miden-lib/src/transaction/inputs.rs

```diff
@@ -1,52 +1,139 @@
 use thiserror::Error;
 
 use super::TransactionKernel;
 
-// TRANSACTION ADVICE INPUTS
+// TRANSACTION KERNEL INPUTS
 // ================================================================================================
 
-/// Advice inputs wrapper for inputs that are meant to be used exclusively in the transaction
-/// kernel.
-#[derive(Debug, Clone, Default)]
-pub struct TransactionAdviceInputs(AdviceInputs);
+/// Contains the data required to execute a transaction.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TransactionKernelInputs {
+    prep_inputs: TransactionPreparationInputs,
+    input_notes: InputNotes<InputNote>,
+    tx_args: TransactionArgs,
+    init_advice_inputs: Option<AdviceInputs>,
+}
+
+impl TransactionKernelInputs {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns new [`TransactionKernelInputs`] instantiated with the specified parameters.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - The partial blockchain does not track the block headers required to prove inclusion of any
+    ///   authenticated input note.
+    pub fn new(
+        prep_inputs: TransactionPreparationInputs,
+        input_notes: InputNotes<InputNote>,
+        tx_args: TransactionArgs,
+        init_advice_inputs: Option<AdviceInputs>,
+    ) -> Result<Self, TransactionInputError> {
+        // Validate the authentication paths of the input notes.
+        for note in input_notes.iter() {
+            if let InputNote::Authenticated { note, proof } = note {
+                let note_block_num = proof.location().block_num();
+                let block_header = if note_block_num == prep_inputs.block_header().block_num() {
+                    prep_inputs.block_header()
+                } else {
+                    prep_inputs.blockchain().get_block(note_block_num).ok_or(
+                        TransactionInputError::InputNoteBlockNotInPartialBlockchain(note.id()),
+                    )?
+                };
+                validate_is_in_block(note, proof, block_header)?;
+            }
+        }
+
+        Ok(Self {
+            prep_inputs,
+            input_notes,
+            tx_args,
+            init_advice_inputs,
+        })
+    }
+
+    // MUTATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Updates the input notes for the transaction.
+    ///
+    /// # Warning
+    ///
+    /// This method does not validate the notes against the data already in this
+    /// [`TransactionKernelInputs`]. It should only be called with notes that have been validated
+    /// by the constructor.
+    pub fn set_input_notes(&mut self, new_notes: Vec<Note>) {
+        self.input_notes = new_notes.into();
+    }
+
+    /// Updates the transaction arguments of the inputs.
+    pub fn set_tx_args(&mut self, tx_args: TransactionArgs) {
```

Comment:
This is used only for testing purposes, right? If so, should we put it behind the `testing` flag?